File: Projects/Dagger/Projects/Project-3-ConstructorInjection/app/src/main/java/com/droid/code/di/components/ApplicationComponent.java
package com.droid.code.di.components;

import com.droid.code.MyApplication;
import com.droid.code.data.local.DatabaseService;
import com.droid.code.data.remote.NetworkService;
import com.droid.code.di.modules.ApplicationModule;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {ApplicationModule.class})
public interface ApplicationComponent {

    void inject(MyApplication application);

    NetworkService getNetworkService();

    DatabaseService getDatabaseService();

}

File: Projects/Dagger/Projects/Project-5-DaggerInKotlin/app/src/main/java/com/demo/kotlin/sample/di/component/ApplicationComponent.kt
package com.demo.kotlin.sample.di.component

import android.content.Context
import com.demo.kotlin.sample.MyApplication
import com.demo.kotlin.sample.data.local.DatabaseService
import com.demo.kotlin.sample.data.remote.NetworkService
import com.demo.kotlin.sample.di.ApplicationContext
import com.demo.kotlin.sample.di.module.ApplicationModule
import com.demo.kotlin.sample.utils.NetworkHelper
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {

    fun inject(application: MyApplication)

    @ApplicationContext
    fun getContext(): Context

    fun getNetworkService(): NetworkService

    fun getDatabaseService(): DatabaseService

    fun getNetworkHelper(): NetworkHelper
}
